src/core/util/props.js

```diff
@@ -166,7 +167,12 @@ function assertType (value: any, type: Function): {
   } else if (expectedType === 'Array') {
     valid = Array.isArray(value)
   } else {
-    valid = value instanceof type
+    try {
+      valid = value instanceof type
+    } catch (e) {
+      warn('Invalid prop type: "' + String(type) + '" is not a constructor', vm);
+      valid = false;
```

Comment:
```suggestion
      warn(`Invalid prop type: "${type}" is not a constructor`, vm)
      valid = false
```
Is the `String` call necessary?

Reply:
Using String is more robust while the interpolation fails for some types like Symbol (which doesn't apply here)